src/exception/error-handling.ts

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2023 The Ontario Institute for Cancer Research. All rights reserved
+ *
+ * This program and the accompanying materials are made available under the terms of
+ * the GNU Affero General Public License v3.0. You should have received a copy of the
+ * GNU Affero General Public License along with this program.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY
+ * EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import { RepoError } from './repo/types';
+import { Errors } from '../utils';
+
+// types
+export type Success<T> = { success: true; exception: T };
+export type Failure = { success: boolean; message: string; errors?: any };
```

Comment:
just to clarify: for type `Success`, the property `success` is always true, but for type `Failure` it could be either true or false?

Reply:
nice catch! changed to a `boolean` from the constant `true`
types should be boolean , then the helper function sets as `true` or `false`

Reply:
That definitely works! Alternatively, one may have `true` and the other `false` as their single expected value, right?
I cannot imagine a successful Failure in this context (while I can in other more ridiculous cases 😆)

Reply:
definitely works too for this. not relevant to this PR specifically. just an aside...

Avoiding the `true` explicit value because in the future there may be cases where one of these types takes something that has been cast into a boolean. I'm unclear if TS is ok with that or needs an explicit `true`. So this at least has that little bit of freedom.  I do see your point though.

Reply:
Agreed on all above! 🙏